### PR TITLE
feat(core): Add agent-eval rating persistence with edit capture (no-changelog)

### PR DESCRIPTION
**Pull request by DeveloperTheExplorer**
## Summary

Records a human's judgment of an agent-eval result — a 👍/👎 vote, an optional comment, and an optional **correction** (the edited answer the agent should have given) — plus the reads a later judge-calibration pass needs. Backend only, behind the `101_agent_evals` rollout flag.

New `AgentEvalRatingService` in `packages/cli/src/modules/agent-evals/`:

- `rateResult` — vote + optional comment/correction against a result
- `listRatingsForResult` — the per-case rating history
- `listLatestRatingsForRun` — newest rating per result across a run

The entity, repository, migration, and DTOs all landed with the data-model PR (#34657), so the net-new persistence here is just `AgentEvalRatingRepository.findLatestByRunId` and `AgentEvalResultRepository.findById`.

Decisions worth a reviewer's attention:

- **The correction is stored on the rating row only.** It is deliberately *not* written back to the dataset row as the case's expected output, so an edit does not become the reference answer for later runs. That product decision is still open; the deferral — and its implication for the review view's copy — is documented on TRUST-293.
- **Ratings are append-only.** Re-voting inserts a new row rather than overwriting, so a flipped judgment stays auditable, which is why the run-level read returns newest-per-result. (Contrast `execution_annotations`, which upserts a single vote per execution.)
- **Newest-per-result is reduced in the repository rather than in SQL.** `DISTINCT ON` is Postgres-only, and a correlated `MAX(createdAt)` returns both rows when two ratings share a millisecond.
- **Authorization** resolves `result → run → dataset → agent` and checks the project scope *before* any lookup (`agent:update` to rate, `agent:read` to read), so a caller without access can't probe which result or run ids exist, and a run in another project reads as missing rather than forbidden.
- **The request body is bounded in the service** (comment 2 000 chars, `correction.finalText` 20 000, whole correction 32 000) because this service — not a controller — is currently the enforcement point; the zod schema carries no `max()`. It rejects rather than truncates: a silently clipped gold answer defeats the purpose.
- **Flag gate is per-user PostHog, throwing not-found when off** so a flag-off instance leaks no flag state. This differs from the sibling runner service, which gates on the env override and whose own comment marks that as temporary pending the REST layer.

Out of scope, by design: REST endpoints (TRUST-290, in progress in parallel — it owns the `@ProjectScope` decorators and the `AgentEvalRatingRecord` mapping), UI (TRUST-293), telemetry (TRUST-295), judging and calibration (TRUST-297 / TRUST-298).

## How to test

There is no HTTP surface yet, so this is exercised through tests:

```bash
cd packages/@n8n/db && pnpm test src/repositories/__tests__/agent-eval-rating.repository.test.ts
cd packages/cli && pnpm test src/modules/agent-evals
cd packages/cli && pnpm test:integration src/modules/agent-evals
```

Results on this branch: 13/13 (`@n8n/db`), 73/73 (cli unit, 15 new), 7/7 (cli integration, 5 new).

The integration test runs `findLatestByRunId` against a real SQLite database with a real project, agent, dataset, run, and seeded results — a stubbed entity manager would green-light SQL the driver rejects. It covers: a re-voted result yielding only the newest rating while the superseded correction stays on record, one rating per rated result with unrated results skipped, correction JSON round-tripping unchanged, run scoping, and a run with no ratings. Ratings are deliberately spaced 10 ms apart because `createdAt` has millisecond precision and the id tie-break is a random UUID.

For a deployed instance once the REST layer lands, the feature needs `N8N_ENABLED_MODULES=agents,agent-evals,data-table` (`agent-evals` is opt-in, not a default module) plus either the `101_agent_evals` PostHog flag or the `N8N_AGENT_EVALS_ENABLED=true` override.

## Related Linear tickets, Github issues, and Community forum posts

- https://linear.app/n8n/issue/TRUST-289
- Gold-answer deferral documented on https://linear.app/n8n/issue/TRUST-293
- Builds on https://linear.app/n8n/issue/TRUST-284 (#34657) and https://linear.app/n8n/issue/TRUST-286 (#34904)
- Consumed by https://linear.app/n8n/issue/TRUST-290

## Review / Merge checklist

- [x] I have seen this code, I have run this code, and I take responsibility for this code.
- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md))
- [x] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included.
- [ ] PR Labeled with `Backport to Beta`, `Backport to Stable`, or `Backport to v1` (if the PR is an urgent fix that needs to be backported)

---

<a href="https://stagereview.app/n8n-io/n8n/pull/35252">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://stagereview.app/assets/gh-open-in-stage-dark.svg">
    Open in Stage
  </picture>
</a>

<a href="https://cubic.dev/pr/n8n-io/n8n/pull/35252?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/@n8n/db/src/repositories/agent-eval-rating.repository.ee.ts [hunk 1]
```diff
@@ -34,4 +37,43 @@ export class AgentEvalRatingRepository extends Repository<AgentEvalRating> {
...
+	async findLatestByRunId(runId: string): Promise<AgentEvalRating[]> {
+		// Ids only — whole rows would load every superseded `correction` just to drop it.
+		const rows = await this.createQueryBuilder('rating')
+			.select('rating.id', 'id')
+			.addSelect('rating.resultId', 'resultId')
+			.innerJoin(AgentEvalResult, 'result', 'result.id = rating.resultId')
+			.where('result.runId = :runId', { runId })
+			.orderBy('rating.resultId', 'ASC')
+			.addOrderBy('rating.createdAt', 'DESC')
+			.addOrderBy('rating.id', 'DESC')
```
Comment (schrothbn) [inline packages/@n8n/db/src/repositories/agent-eval-rating.repository.ee.ts:54]: Why do we order by ID here? Wouldn't that just be as random?

File: packages/@n8n/db/src/repositories/__tests__/agent-eval-rating.repository.test.ts [hunk 2]
```diff
@@ -66,4 +66,51 @@ describe('AgentEvalRatingRepository', () => {
...
+				{ id: 'r-3', resultId: 'res-2' },
+				{ id: 'r-2', resultId: 'res-1' },
+			] as AgentEvalRating[]);
+
+			const latest = await repo.findLatestByRunId('run-1');
+
+			expect(qb.where).toHaveBeenCalledWith('result.runId = :runId', { runId: 'run-1' });
+			// The superseded r-1 is never fetched, only the two winners.
+			expect(entityManager.find.mock.calls[0]?.[1]).toEqual({
+				where: { id: expect.objectContaining({ _value: ['r-2', 'r-3'] }) },
```
Comment (schrothbn) [inline packages/@n8n/db/src/repositories/__tests__/agent-eval-rating.repository.test.ts:104]: Probably would be enough to test that in the integration test instead of asserting on internal values of type orm which may or may not change at some point, breaking this on an update for no real reason. Edge case of course.

**Review by schrothbn — APPROVED** [review]
Looks fine.
Maybe we can remove the unit test asserting typeorm internals as it's probably covered better in an integration test and one question to the ordering, nothing that would block.

File: packages/@n8n/db/src/repositories/agent-eval-rating.repository.ee.ts (hunk 1 above)
Comment (DeveloperTheExplorer) [reply packages/@n8n/db/src/repositories/agent-eval-rating.repository.ee.ts:54]: It's mostly to remove randomness and bring stability when plans change and have a tiebreaker.

File: packages/@n8n/db/src/repositories/__tests__/agent-eval-rating.repository.test.ts (hunk 2 above)
Comment (DeveloperTheExplorer) [reply packages/@n8n/db/src/repositories/__tests__/agent-eval-rating.repository.test.ts:104]: noted, fill fix in follow up PRs
